src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -934,6 +1116,205 @@ - (void)readAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
     std::move(*bridge).DispatchAction(self);
 }
 
+- (void)subscribeWithAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullable)attributePaths
```

Comment:
Again, the existing subscribeToAttributes/Events bits should just call this one.  And this should be placed right next to where the current subscribeToAttributesWithEndpointID implementation is, to make the diff saner to review.

Reply:
I understand.